esp-backtrace/src/riscv.rs

```diff
@@ -12,45 +12,93 @@ pub(super) const RA_OFFSET: usize = 4;
 
 /// Registers saved in trap handler
 #[doc(hidden)]
-#[allow(missing_docs)]
 #[derive(Default, Clone, Copy)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[repr(C)]
 pub(crate) struct TrapFrame {
+    /// Return address, stores the address to return to after a function call or
+    /// interrupt.
     pub ra: usize,
+    /// Temporary register t0, used for intermediate values.
```

Comment:
This is terrible, can we turn these into arrays?

Reply:
I agree, but this is not the subject of this PR, so I would not like to make any changes here in terms of this commit tbh.